crates/ruff/src/checkers/tokens.rs

```diff
@@ -184,9 +191,26 @@ pub(crate) fn check_tokens(
     }
 
     // TD001, TD002, TD003, TD004, TD005, TD006, TD007
+    // T001, T002, T003, T004
     if enforce_todos {
+        let todo_comments: Vec<TodoComment> = indexer
+            .comment_ranges()
+            .iter()
+            .enumerate()
+            .filter_map(|(i, comment_range)| {
+                let comment = locator.slice(*comment_range);
+                TodoComment::from_comment(comment, comment_range, i)
+            })
+            .collect::<Vec<TodoComment>>();
```

Comment:
Nit: It's sufficient to either annotate the type of the `tood_comments` variable or specify the type argument when calling `collect`
```suggestion
        let todo_comments: Vec<TodoComment> = indexer
            .comment_ranges()
            .iter()
            .enumerate()
            .filter_map(|(i, comment_range)| {
                let comment = locator.slice(*comment_range);
                TodoComment::from_comment(comment, comment_range, i)
            })
            .collect();
```